### PR TITLE
Migrate Arrays in FSharp.Core reference

### PR DESCRIPTION
**Pull request by cartermp**
In service towards https://github.com/dotnet/docs/issues/17199

### CODE REVIEW

**Comment by cartermp** [comment]
@mairaw Would you happen to know why the TOC for `/arrays` doesn't render in the staging site? Example: https://review.docs.microsoft.com/en-us/dotnet/fsharp/library-reference/core-library-reference/array/array.append['t]-function?branch=pr-en-us-17254

File: docs/fsharp/library-reference/core-library-reference/array/toc.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        - name: Array.Parallel.iteri<'T> Function
+          href: parallel/parallel.iteri['t]-function.md
+        - name: Array.Parallel.map<'T,'U> Function
+          href: parallel/parallel.map['t,'u]-function.md
+        - name: Array.Parallel.mapi<'T,'U> Function
+          href: parallel/parallel.mapi['t,'u]-function.md
+        - name: Array.Parallel.parition<'T> Function
+          href: parallel/parallel.partition['t]-function.md
+    - name: Array.append<'T> Function
+      href: array.append['t]-function.md
```
Comment (mairaw) [inline docs/fsharp/library-reference/core-library-reference/array/toc.yml]: Try this:

```suggestion
      href: array.append%5B't%5D-function.md
```

This is how GitHub renders that file name

File: docs/fsharp/library-reference/core-library-reference/array/array.append['t]-function.md [hunk 2]
```diff
@@ -24,8 +24,8 @@ Array.append array1 array2
 
 |Paramter name|Type|Description|
```
Comment (mairaw) [inline docs/fsharp/library-reference/core-library-reference/array/array.append['t]-function.md]:
```suggestion
|Parameter name|Type|Description|
```

**Comment by mairaw** [comment]
Meh, it doesn't seem to work. I'd reach out to Docs support channel. Are you familiar with that on Teams?

**Comment by cartermp** [comment]
Looks like it's related to the level of nesting

**Comment by cartermp** [comment]
Nope, unrelated to nesting I guess. I'm sort of on the verge of giving up here. Is there no way to just generate a compliant TOC based on file structure?

**Pull request merged**